### ISSUE TITLE
Implement unified interface for historical course data

### ISSUE THREAD

**Issue by Zvezdin** [knowm/XChange#1688]
The poloniex api allows you to download historical course data for different intervals (candlestick data each 5 min for example).

Example API call:
`https://poloniex.com/public?command=returnChartData&currencyPair=USDT_ETH&start=1499603961&end=9999999999&period=300`

The Poloniex API wrapper in XChange implements this feature, but it can only be used by writing exchange-specific code (Poloniex for this case), which isn't very flexible. Do you consider implementing a unified exchange-independent way to download historical data from the exchanges that support it?

**Comment by timmolter** [comment]
Yes, that'd be fine. It should be implemented similar to the `Trades` code.
